kong/llm/drivers/azure.lua

```diff
@@ -13,6 +13,18 @@ local string_gsub = string.gsub
 local DRIVER_NAME = "azure"
 --
 
+local do_not_clear_headers = {
+  ["host"] = true,
+  ["content-length"] = true,
+  ["transfer-encoding"] = true,
+  ["connection"] = true,
+  ["accept-encoding"] = true,
+  ["content-type"] = true,
+  ["accept"] = true,
+  ["user-agent"] = true,
```

Comment:
There are many auth headers that also should not be cleared, as well as any Bedrock signed headers.

This change worries me.